tests/TestContextTest.cs

```diff
@@ -0,0 +1,186 @@
+        public async Task Test004()
+        {
+            using (var target = CreateTestContext())
+            {
+                var asyncTestDepMock = new Mock<IAsyncTestDep>();
+                var getDataTask = new TaskCompletionSource<string>();
+                asyncTestDepMock.Setup(o => o.GetData()).Returns(getDataTask.Task);
+                target.Services.AddService<IAsyncTestDep>(asyncTestDepMock.Object);
+
+                var cut = target.RenderComponent<SimpleWithAyncDeps>();
+                var renderTask = target.WaitForNextRender(NoOperation);
+
+                asyncTestDepMock.Verify(o => o.GetData());
+                renderTask.IsCompleted.ShouldBeFalse();
+
+                getDataTask.SetResult("Hello, World!");
+                await renderTask;
+
+                renderTask.IsCompleted.ShouldBeTrue();
+            }
+        }
+
+        [Fact(DisplayName = "No render when asynchronous initialization fails")]
+        public async Task Test005()
+        {
+            using (var target = CreateTestContext())
+            {
+                var asyncTestDepMock = new Mock<IAsyncTestDep>();
+                var getDataTask = new TaskCompletionSource<string>();
+                asyncTestDepMock.Setup(o => o.GetData()).Returns(getDataTask.Task);
+                target.Services.AddService<IAsyncTestDep>(asyncTestDepMock.Object);
+
+                var cut = target.RenderComponent<SimpleWithAyncDeps>();
+                
+                asyncTestDepMock.Verify(o => o.GetData());
+
+                var renderTask = target.WaitForNextRender(
+                    () => getDataTask.SetException(new InvalidOperationException()),
+                    TimeSpan.FromMilliseconds(100));
+
+                renderTask.IsCompleted.ShouldBeFalse();
+
+                await Should.ThrowAsync<TimeoutException>(renderTask);
+            }
+        }
+
+        [Fact(DisplayName = "Wait for render after asynchronous event")]
+        public async Task Test006()
+        {
+            using (var target = CreateTestContext())
+            {
+                var asyncTestDepMock = new Mock<IAsyncTestDep>();
+                asyncTestDepMock.Setup(o => o.GetData()).ReturnsAsync(string.Empty);
+                target.Services.AddService<IAsyncTestDep>(asyncTestDepMock.Object);
+
+                var cut = target.RenderComponent<SimpleWithAyncDeps>();
+
+                var getDataTask = new TaskCompletionSource<string>();
+                asyncTestDepMock.Setup(o => o.GetData()).Returns(getDataTask.Task);
+                asyncTestDepMock.Invocations.Clear();
+
+                var button = cut.Find("button");
+                var renderTask = target.WaitForNextRender(() => button.Click());
+
+                // Render is done on button click
+                asyncTestDepMock.Verify(o => o.GetData());
+                renderTask.IsCompleted.ShouldBeTrue();
+                await renderTask;
```

Comment:
This part of the test is not entirely clear to me. `button.Click()` is synchronous and blocks until the render is done, so is this part relevant?

Could this be replaced by

```suggestion
cut.Find("button").Click();
```